app/src/main/java/com/termux/app/terminal/io/extrakeys/ExtraKeysView.java

```diff
@@ -0,0 +1,388 @@
+
+        if (!state.isOn || !state.isActive)
+            return false;
+
+        state.setIsActive(false);
+
+        return true;
+    }
+
+    private Button createSpecialButton(String buttonKey, boolean needUpdate) {
+        SpecialButtonState state = specialButtons.get(SpecialButton.valueOf(buttonKey));
+        if (state == null) return null;
+        state.isOn = true;
+        Button button = new Button(getContext(), null, android.R.attr.buttonBarButtonStyle);
+        button.setTextColor(state.isActive ? INTERESTING_COLOR : TEXT_COLOR);
+        if (needUpdate) {
+            state.buttons.add(button);
+        }
+        return button;
+    }
+
+    void popup(View view, ExtraKeyButton extraButton) {
+        int width = view.getMeasuredWidth();
+        int height = view.getMeasuredHeight();
+        Button button;
+        if (isSpecialButton(extraButton)) {
+            button = createSpecialButton(extraButton.getKey(), false);
+            if (button == null) return;
+        } else {
+            button = new Button(getContext(), null, android.R.attr.buttonBarButtonStyle);
+            button.setTextColor(TEXT_COLOR);
+        }
+        button.setText(extraButton.getDisplay());
+        button.setPadding(0, 0, 0, 0);
+        button.setMinHeight(0);
+        button.setMinWidth(0);
+        button.setMinimumWidth(0);
+        button.setMinimumHeight(0);
+        button.setWidth(width);
+        button.setHeight(height);
+        button.setBackgroundColor(BUTTON_PRESSED_COLOR);
+        popupWindow = new PopupWindow(this);
+        popupWindow.setWidth(LayoutParams.WRAP_CONTENT);
+        popupWindow.setHeight(LayoutParams.WRAP_CONTENT);
+        popupWindow.setContentView(button);
+        popupWindow.setOutsideTouchable(true);
+        popupWindow.setFocusable(false);
+        popupWindow.showAsDropDown(view, 0, -2 * height);
+    }
+
+    /**
+     * General util function to compute the longest column length in a matrix.
+     */
+    static int maximumLength(Object[][] matrix) {
+        int m = 0;
+        for (Object[] row : matrix)
+            m = Math.max(m, row.length);
+        return m;
+    }
+
+    /**
+     * Reload the view given parameters in termux.properties
+     *
+     * @param infos matrix as defined in termux.properties extrakeys
+     * Can Contain The Strings CTRL ALT TAB FN ENTER LEFT RIGHT UP DOWN or normal strings
+     * Some aliases are possible like RETURN for ENTER, LT for LEFT and more (@see controlCharsAliases for the whole list).
+     * Any string of length > 1 in total Uppercase will print a warning
+     *
+     * Examples:
+     * "ENTER" will trigger the ENTER keycode
+     * "LEFT" will trigger the LEFT keycode and be displayed as "←"
+     * "→" will input a "→" character
+     * "−" will input a "−" character
+     * "-_-" will input the string "-_-"
+     */
+    @SuppressLint("ClickableViewAccessibility")
+    public void reload(ExtraKeysInfo infos) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 28. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.